Lib/asyncio/queues.py

```diff
@@ -112,7 +127,11 @@ async def put(self, item):
 
         Put an item into the queue. If the queue is full, wait until a free
         slot is available before adding item.
+
+        Raises QueueShutDown if the queue has been shut down.
         """
+        if self._is_shutdown:
+            raise QueueShutDown
         while self.full():
```

Comment:
```suggestion
        while self.full():
            if self._is_shutdown:
                raise QueueShutDown
```

I think I can reduce duplication with this and the next suggestion

Reply:
This didn't happen -- are you still planning this change?

Reply:
If you run that example:
```py
async def _test_shutdown_notempty_get():
    res = []
    q = asyncio.Queue()
    await q.put("xx")
    q.shutdown(False)
    await q.get() # Succeed !!!
```
No `ShutDown` exception raises, that's a problem because queue is shutdown.
See my comment about the `get_nowait` method.

Reply:
No, that code example that you just posted is correct, and is expected to succeed. `shutdown(immediate=False)` is supposed to let consumers empty the queue before excepting